### ISSUE TITLE
`regexp/strict` conflict with `unicorn/better-regex`

### ISSUE THREAD

**Issue by JounQin** [ota-meshi/eslint-plugin-regexp#445]
**Information:**
- ESLint version: v8.19.0
- `eslint-plugin-regexp` version: 1.7.0

**Description**

```js
/\[([^[\]]+)]/g
```

error from this plugin

```log
Unescaped source character ']'
```

After fixed:

```js
/\[([^[\]]+)\]/g
```

error from `unicorn/better-regex`

```log
/\[([^[\]]+)\]/g can be optimized to /\[([^[\]]+)]/g
```

**Comment by JounQin** [comment]
```js
/{{([\S\s]+?)}}/g
```

`regexp/match-any` is reported

```js
/{{([\s\S]+?)}}/g
```

`unicorn/better-regex` is reported

**Comment by JounQin** [comment]
I think we should adapt https://github.com/SamVerschueren/clean-regexp and https://github.com/DmitrySoshnikov/regexp-tree

**Comment by ota-meshi** [comment]
I don't know what the `unicorn/better-regex` says is better, but I think the`regexp/strict` rule is working correctly.

`/]/` is allowed by Annex B, but if we don't use Annex B (We describe it as `strict`), I think it's a regex that is strictly a syntax error.

https://ota-meshi.github.io/eslint-plugin-regexp/rules/strict.html

**Comment by ota-meshi** [comment]
The `regexp/match-any` rule allows you to use options to choose your preferred notation.

https://ota-meshi.github.io/eslint-plugin-regexp/rules/match-any.html

**Comment by JounQin** [comment]
Raised https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1852 to discuss.

I hope these two plugins can work better together by default, some rules can be extracted from `unicorn` to `regexp` instead?

**Comment by ota-meshi** [comment]
In my opinion, if you want to use `unicorn/better-regex`, recommend not to use `eslint-plugin-regexp`.
If you want to use `eslint-plugin-regexp`, recommended not to use `unicorn/better-regex`.

`unicorn/better-regex` can be used without complicated options. However, detailed user preferences cannot be set. `eslint-plugin-regexp` has many options, so you can configure it to your preferences.
I think they have different purposes.

**Comment by JounQin** [comment]
I think the default options should be best practices for most projects, so although these are two different plugins, but they can share why they choose different style, and what's the benefits and downsides, and then we can choose whether or not change the default behaviors.

**Comment by ota-meshi** [comment]
I don't know what the idea of `unicorn/better-regex` is to say "better". So I don't know if that makes sense. If we're not sure if that regexp is "better", it doesn't make sense to change the default behavior.
Do you know about the "better" of `unicorn/better-regex`?

**Comment by JounQin** [comment]
> Raised [sindresorhus/eslint-plugin-unicorn#1852](https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1852) to discuss.
> 
> I hope these two plugins can work better together by default, some rules can be extracted from `unicorn` to `regexp` instead?

That's why I raised the related issue for discussion with maintainers from `eslint-plugin-unicorn`, we can wait for a while for them?

**Comment by yakov116** [comment]
`better-regex` was dropped from unicorn

**Comment by fisker** [comment]
no... that's `safe-regex`

**Comment by fisker** [comment]
`unicorn/better-regex` is now dead.
 
https://github.com/sindresorhus/eslint-plugin-unicorn/releases/tag/v65.0.0

**Comment by ota-meshi** [comment]
Thank you for letting me know!
